nautobot_ssot_infoblox/diffsync/adapters/infoblox.py

```diff
@@ -118,18 +125,24 @@ def load_vlans(self):
 
     def load(self):
         """Load all models by calling other methods."""
-        self.load_prefixes()
-        if "prefix" in self.dict():
-            self.job.log(message=f"Loaded {len(self.dict()['prefix'])} prefixes from Infoblox.")
-        self.load_ipaddresses()
-        if "ipaddress" in self.dict():
-            self.job.log(message=f"Loaded {len(self.dict()['ipaddress'])} IP addresses from Infoblox.")
-        self.load_vlanviews()
-        if "vlangroup" in self.dict():
-            self.job.log(message=f"Loaded {len(self.dict()['vlangroup'])} VLAN views from Infoblox.")
-        self.load_vlans()
-        if "vlan" in self.dict():
-            self.job.log(message=f"Loaded {len(self.dict()['vlan'])} VLANs from Infoblox.")
+        if "import_objects" in PLUGIN_CFG:
+            if PLUGIN_CFG["import_objects"].get("subnets"):
+                self.load_prefixes()
+            if PLUGIN_CFG["import_objects"].get("ip_addresses"):
+                self.load_ipaddresses()
+            if PLUGIN_CFG["import_objects"].get("vlan_views"):
+                self.load_vlanviews()
+            if PLUGIN_CFG["import_objects"].get("vlans"):
+                self.load_vlans()
+        else:
+            self.job.log_info(message="The `import_objects` setting was not found so all objects will be imported.")
+            self.load_prefixes()
+            self.load_ipaddresses()
+            self.load_vlanviews()
+            self.load_vlans()
+        for obj in ["prefix", "ipaddress", "vlangroup", "vlan"]:
```

Comment:
maybe, for easy navigation, the obj name should match the ones from the config? i.e. `ip_addresses` vs `ipaddress`